lib/plugins/wallet/cardano/cardano.js

```diff
@@ -0,0 +1,85 @@
+const { walletId } = require('./mnemonic')
+const cardanoApi = require('./client')
+
+// TODO: work with BN here instead of api.js?
+
+async function checkCryptoCode(cryptoCode) {
+  if (cryptoCode !== 'ADA') {
+    return Promise.reject(new Error(`Unsupported crypto: ${cryptoCode}`))
+  }
+
+  return Promise.resolve()
+}
+
+/**
+ * TODO: instead of using walletId function we can
+ *       inject walletId/index into account object
+ *       in wallet.js - it's better?
+ */
+async function balance(account, cryptoCode) {
+  return checkCryptoCode(cryptoCode)
+    .then(() => walletId())
+    .then(({ walletId }) => cardanoApi.walletBalance(walletId, 1))
+}
+
+async function sendCoins(account, address, cryptoAtoms, cryptoCode) {
+  return checkCryptoCode(cryptoCode)
+    .then(() => walletId())
+    .then(({ walletId, index }) =>
+      cardanoApi.sendCoins(walletId, index, address, cryptoAtoms)
+    )
+}
+
+async function newAddress(account, info) {
+  return checkCryptoCode(info.cryptoCode)
+    .then(() => walletId())
+    .then(({ walletId, index }) => cardanoApi.newAddress(walletId, index))
+}
+
+async function getStatus(account, toAddress, requested, cryptoCode) {
+  return checkCryptoCode(cryptoCode)
+    .then(() =>
+      Promise.all([
+        cardanoApi.addressBalance(toAddress, 1),
```

Comment:
is `1` a good constant for confirmed balance? I think it
1) might need to be more
2) should a named constant

Reply:
they used `1` for all BTC like cryptocurrencies, but we may change this - have you idea for good count of confirmations? Daedalus use 1-2 for low, 3-8 for medium and 9+ for high assurance. Constant is a good idea